### PR TITLE
Fix missing textures in atlas.

### PR DESCRIPTION
**Pull request by bagnell**
Render to texture instead of copy from framebuffer on texture atlas resize. For #2997.

Still investigating GL calls for submitting a Chrome bug.

### CODE REVIEW

File: Source/Renderer/RenderState.js [hunk 1]
```diff
@@ -422,6 +422,23 @@ define([
         return cachedState;
     };
 
+    /**
+     * Use sparingly until we have a real solution for removing render states from the cache, e.g. TextureAtlas
+     * polutting the cache on resize.
+     *
+     * @private
+     */
+    RenderState.removeFromCache = function(renderState) {
```
Comment (pjcozzi) [inline Source/Renderer/RenderState.js]: We need to reference count these, right?  What if someone else created the same render state?

**Comment by pjcozzi** [comment]
Update CHANGES.md.

File: Source/Scene/TextureAtlas.js [hunk 2]
```diff
@@ -201,16 +223,25 @@ define([
...
+            });
+
+            var command = textureAtlas._copyCommand;
+            command.renderState = RenderState.fromCache({
+                viewport : new BoundingRectangle(0, 0, oldAtlasWidth, oldAtlasHeight)
             });
+
             framebuffer._bind();
-            newTexture.copyFromFramebuffer(0, 0, 0, 0, oldAtlasWidth, oldAtlasHeight);
+            command.execute(textureAtlas._context);
```
Comment (pjcozzi) [inline Source/Scene/TextureAtlas.js]: `TextureAtlas` is some of the oldest code in Cesium - it predates commands by at least a year.

I am not sure that `TextureAtlas` should keep a reference to `context`.  No one else does that.  This allows `TextureAtlas` to make WebGL calls whenever it wants regardless of the global state.  This is not good for scheduling `requestAnimFrame` and for interop for other WebGL engines (#648).  I wonder if it is even the source of this issue.

Perhaps we could replace `_context` with an `update` function and use `ComputeCommand`s?  How hard is that refactor?  Is it too architect astronaut-ish?

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look.

I agree that `TextureAtlas` should have an `update` function, but we should create an issue and add it in the future.

**Comment by pjcozzi** [comment]
Works well and code looks good.  Just add unit tests to RenderStateSpec.js.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look. I changed how the states are reference counted. The partial keys are reference counted as before, but the full key reference count is now how many partial keys reference the full state.

**Comment by pjcozzi** [comment]
Looks good.

**Pull request merged**